earthmover/yaml_parser.py

```diff
@@ -30,6 +30,33 @@ def update(self, _dict: 'YamlMapping'):
                 self[key] = val
         return self
 
+    # enables setting values inside the YamlMapping by path, e.g. `configs.set_path("config.tmp_dir", "/mypath")`
+    def set_path(self, path, value):
+        path_pieces = path.split(".")
+        current = self
+        for path_piece in path_pieces[:-1]:
+            if path_piece not in current.keys():
+                current[path_piece] = YamlMapping()
+            current = current[path_piece]
+        current[path_pieces[-1]] = self.autocast(value)
```

Comment:
This works because `current` is a _reference_, so the value in self also gets updated. `current[path_piece] = YamlMapping()` creates keys if they don't exist, to prevent a `KeyError`.